core/src/main/java/org/apache/accumulo/core/file/rfile/GenerateSplits.java

```diff
@@ -200,14 +202,21 @@ private List<UnreferencedTabletFile> getFiles(FileSystem fs, Path path) throws I
 
   private Text[] getQuantiles(SortedKeyValueIterator<Key,Value> iterator, int numSplits)
       throws IOException {
-    ItemsSketch<Text> itemsSketch = ItemsSketch.getInstance(BinaryComparable::compareTo);
+    var itemsSketch = ItemsSketch.getInstance(Text.class, BinaryComparable::compareTo);
     while (iterator.hasTop()) {
       Text row = iterator.getTopKey().getRow();
       itemsSketch.update(row);
       iterator.next();
     }
-    Text[] items = itemsSketch.getQuantiles(numSplits + 2);
-    // based on the ItemsSketch javadoc, method returns min, max as well so drop first and last
+    // the number requested represents the number of regions between the resulting array elements
+    // the actual number of array elements is one more than that to account for endpoints;
+    // so, we ask for one more because we want the number of median elements in the array to
+    // represent the number of split points and we will drop the first and last array element
+    double[] ranks = QuantilesUtil.equallyWeightedRanks(numSplits + 1);
+    // the choice to use INCLUSIVE or EXCLUSIVE is arbitrary here; EXCLUSIVE matches the behavior
+    // of datasketches 3.x, so we might as well preserve that for 4.x
+    Text[] items = itemsSketch.getQuantiles(ranks, QuantileSearchCriteria.EXCLUSIVE);
+    // drop the min and max, so we only keep the median elements to use as split points
```

Comment:
The following would also work equivalently to these two lines, but given how confusing the API is, I think it's better to explain things in two steps. These two steps are done internally to this one line anyway plus some other stuff we don't need, so what I have in two lines, with comments, is the minimum work to get the job done anyway.

```java
    Text[] items = ItemsSketch.html#getPartitionBoundaries(numSplits + 1, QuantileSearchCriteria.EXCLUSIVE).boundaries;
```